src/app/analytics/analytics.factory.js

```diff
@@ -278,540 +292,479 @@ const analyticsFactory = /* @ngInject */ function ($window, $timeout, envService
-        // Error caught in analyticsFactory.getSetProductCategory
-      }
-    },
-    giveGiftModal: function (productData) {
-      try {
-        const product = [{
-          productInfo: {
-            productID: productData.designationNumber
-          },
-          attributes: {
-            siebel: {
-              producttype: 'designation'
+        if (desigType !== null) {
+          const productConfig = $window.document.getElementsByTagName('product-config')
+          $window.digitalData.product = [{
+            productInfo: {
+              productID: productConfig.length ? productConfig[0].getAttribute('product-code') : null
+            },
+            category: {
+              primaryCategory: 'donation ' + desigType ? desigType.toLowerCase() : '',
+              siebelProductType: 'designation',
+              organizationId: path[0]
             }
-          }
-        }]
+          }]
 
-        $window.digitalData.product = product
-        $window.dataLayer.push({
-          event: 'give-gift-modal',
-          ecommerce: {
-            currencyCode: 'USD',
-            detail: {
-              products: [{
-                name: productData.designationNumber,
-                id: productData.designationNumber,
-                price: undefined,
-                brand: productData.orgId,
-                category: productData.designationType.toLowerCase(),
-                variant: undefined,
-                quantity: '1'
-              }]
-            }
-          }
-        })
-        this.setEvent('give gift modal')
-        this.pageLoaded()
-      } catch (e) {
-        // Error caught in analyticsFactory.giveGiftModal
+          return path[0]
+        }
       }
-    },
-    pageLoaded: function (skipImageRequests) {
-      try {
-        const path = this.getPath()
-        this.getSetProductCategory(path)
-        this.setSiteSections(path)
-        this.setLoggedInStatus()
-
-        if (typeof $window.digitalData.page.attributes !== 'undefined') {
-          if ($window.digitalData.page.attributes.angularLoaded === 'true') {
-            $window.digitalData.page.attributes.angularLoaded = 'false'
-          } else {
-            $window.digitalData.page.attributes.angularLoaded = 'true'
+
+      return false
+    }),
+    giveGiftModal: suppressErrors(function (productData) {
+      const product = [{
+        productInfo: {
+          productID: productData.designationNumber
+        },
+        attributes: {
+          siebel: {
+            producttype: 'designation'
           }
-        } else {
-          $window.digitalData.page.attributes = {
-            angularLoaded: 'true'
+        }
+      }]
+      $window.digitalData.product = product
+      $window.dataLayer = $window.dataLayer || []
+      $window.dataLayer.push({
+        event: 'give-gift-modal',
+        ecommerce: {
+          currencyCode: 'USD',
+          detail: {
+            products: [generateProduct(productData)]
```

Comment:
`Price` isn't on the `productData` object. it will already be set as undefined. However the items that set `variant` are.

I'll create a new object from  `productData` and will modify it to cause `variant` to be undefined.
```
const modifiedProductData = {...productData}
modifiedProductData.frequency = undefined
      ```